src/schema.rs

```diff
@@ -63,11 +73,22 @@ impl RootDataType {
 
     /// Convert into an Arrow schema.
     pub fn create_arrow_schema(&self, user_metadata: &HashMap<String, String>) -> Schema {
+        self.create_arrow_schema_with_options(user_metadata, TimestampPrecision::default())
+    }
+
+    /// Convert into an Arrow schema with specified timestamp precision.
+    pub fn create_arrow_schema_with_options(
+        &self,
+        user_metadata: &HashMap<String, String>,
+        timestamp_precision: TimestampPrecision,
+    ) -> Schema {
```

Comment:
This should take a structure (built with a Builder) so that 1. adding a new option is not a breaking change  2. options have a default value.